src/pymor/operators/symplectic.py

```diff
@@ -0,0 +1,21 @@
+# This file is part of the pyMOR project (https://www.pymor.org).
+# Copyright pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+
+from pymor.operators.block import BlockOperator
+from pymor.operators.constructions import IdentityOperator
+
+
+class CanonicalSymplecticFormOperator(BlockOperator):
+    """An |Operator| for a canonically symplectic form.
+
+    Parameters
+    ----------
+    half_space
```

Comment:
Why not call it `phase_space`? Is done elsewhere, I believe.

Reply:
Note that `half_space` is only half of the (even-dimensional) phase_space and the Operator inherits from BlockOperator. The typical workflow was to generate the `J`-operator and then use `J.source` as `phase_space` (e.g. `U=J.source.empty()`). This way I get that `J.source == U.space`. When J is generated on-the-fly from a given `phase_space` (which is a `BlockVectorSpace`) how to set the source of the operator elegantly such that `J.source == given_phase_space`? This would probably also become a problem if `J` is generated on-the-fly as suggested in the other reviews? With the current implementation two different `CanonicalSymplecticFormOperator` would be based on the same `half_space` but would generate two different blocked sources in the constructor of `BlockOperatorBase`.